### ISSUE TITLE
ROI : mean spectrum inside a roi

### ISSUE THREAD

**Issue by etienne-monier** [hyperspy/hyperspy#1987]
Hi everyone,

I use HyperSpy for the visualisation of EELS data.

I use this code to:
1. Disp1: display a full 2D signal (for a given band),
2. Disp2: display a roi as a 2D signal (it shows the signal for a given band and it gives the sum of the spectra for this roi).

```
X = np.ones((m,n,l)) # m, n, l are respectively the number of spatial rows, spatial columns and bands (channels)
s = hs.signals.Signal1D(np.moveaxis(X,-1,0)) # X is a numpy array
roi = hs.roi.RectangularROI(left=0,top=0,right=5,bottom=5)
s.plot()
small=roi.interactive(s)
small.plot() 
```

The problem is that if I enlarge the roi inside Disp1, the sum of the spectra of Disp2 get higher. And the curve gets out of the axis so that I can not see it any more.

Solutions: dynamically change the range of the Disp2 spectrum axis or do not display the sum but the mean of the spectra.

I'm just using HyperSpy since this morning, so I may miss something ...

**Comment by thomasaarholt** [comment]
I see your point! I'm afraid I've uncovered a more serious bug in testing for yours. I would normally insert `small_mean = hs.interactive(small.mean, event=roi.events.changed)` before your last line, and replace the final `small` with `small_mean`, but it's not working. I've opened an issue to correct this: #1988.

**Comment by thomasaarholt** [comment]
@Etienne5131 @ericpre has fixed the issue I created with #1994. You can either install the development version from github or wait for the next (imminent) release.

**Comment by ericpre** [comment]
It should be fixed in the release 1.3.2.
